cpp/src/parquet/column_writer.cc

```diff
@@ -1190,7 +1192,7 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
                                   int64_t* out_spaced_values_to_write,
                                   int64_t* null_count) {
     if (bits_buffer_ == nullptr) {
-      if (!level_info_.HasNullableValues()) {
+      if (level_info_.def_level == 0) {
```

Comment:
Interesting. Why is a special case needed for this? Would you like to add a comment?

Reply:
Do you also want to check `def_levels` for null/non-null-ness? (debug mode check perhaps?)

Reply:
I added comments and a check.  I document the method to make it clearer that it always updates the output parameters as well.